boskos/client/client.go

```diff
@@ -71,12 +74,23 @@ type Client struct {
 //
 // Clients created with this function default to retrying failed connection
 // attempts three times with a ten second pause between each attempt.
-func NewClient(owner string, url string) *Client {
+func NewClient(owner string, url, username, password string) *Client {
+
+	getPassword := func() []byte { return []byte("") }
+	if password != "" {
+		sa := &secret.Agent{}
+		if err := sa.Start([]string{password}); err != nil {
+			logrus.WithError(err).Fatal("Failed to start secrets agent")
+		}
+		getPassword = sa.GetTokenGenerator(password)
```

Comment:
Are we sure that the `logrus` logger is updated correctly to censor this?

Reply:
censor setup is in the `sa.Start` function. Add some code for test:

```go
        logrus.Errorf("hello")
	var getPassword func() []byte
	if password != "" {
		sa := &secret.Agent{}
		if err := sa.Start([]string{password}); err != nil {
			logrus.WithError(err).Fatal("Failed to start secrets agent")
		}
		getPassword = sa.GetTokenGenerator(password)
	}

	logrus.Errorf("hello")
```

Test results:

```bash
$ ./boskosctl acquire  --type my-thing --state clean --target-state dirty --owner-name test --server-url http://localhost:8080
ERRO[0000] hello                                        
ERRO[0000] hello                                        
^C
[hongkliu@MiWiFi-R1CM-srv test-infra]$ cat /tmp/password.txt
hello
[hongkliu@MiWiFi-R1CM-srv test-infra]$ ./boskosctl acquire  --type my-thing --state clean --target-state dirty --owner-name test --server-url http://localhost:8080 --username a --password /tmp/password.txt
ERRO[0000] hello                                        
ERRO[0000] CENSORED                                        
^C

```